src/main/java/com/exactpro/th2/sailfish/utils/filter/EqualityFilter.java

```diff
@@ -46,9 +46,9 @@ public FilterOperation getOperation() {
     }
 
     @Override
-    public ExpressionResult validate(Object actualValue) throws RuntimeException {
-        validateActualValue(actualValue);
-        return ExpressionResult.create(value.equals(actualValue) == shouldBeEqual);
+    public ExpressionResult validateInternal(Object value) {
+        validateActualValue(value);
+        return ExpressionResult.create(this.value.equals(value) == shouldBeEqual);
```

Comment:
It is hard to understand which parameter `value` is actual and expected. Please rename the `value` to something readable